official/vision/beta/projects/yolo/modeling/backbones/darknet.py

```diff
@@ -319,86 +505,135 @@ def _csp_stack(self, inputs, config, name):
       csp_filter_scale = 2
       residual_filter_scale = 1
       scale_filters = 2
-    self._default_dict["activation"] = self._get_activation(config.activation)
-    self._default_dict["name"] = f"{name}_csp_down"
-    x, x_route = nn_blocks.CSPRoute(filters=config.filters,
-                                    filter_scale=csp_filter_scale,
-                                    downsample=True,
-                                    **self._default_dict)(inputs)
-    for i in range(config.repetitions):
-      self._default_dict["name"] = f"{name}_{i}"
-      x = nn_blocks.DarkResidual(filters=config.filters // scale_filters,
-                                 filter_scale=residual_filter_scale,
-                                 **self._default_dict)(x)
-
-    self._default_dict["name"] = f"{name}_csp_connect"
-    output = nn_blocks.CSPConnect(filters=config.filters,
-                                  filter_scale=csp_filter_scale,
-                                  **self._default_dict)([x, x_route])
-    self._default_dict["activation"] = self._activation
-    self._default_dict["name"] = None
+    self._default_dict['activation'] = self._get_activation(config.activation)
+    self._default_dict['name'] = f'{name}_csp_down'
+    if self._dilate:
+      self._default_dict['dilation_rate'] = config.dilation_rate
+    else:
+      self._default_dict['dilation_rate'] = 1
+
+    # swap/add dialation
+    x, x_route = nn_blocks.CSPRoute(
+        filters=config.filters,
+        filter_scale=csp_filter_scale,
+        downsample=True,
+        **self._default_dict)(
+            inputs)
+
+    dilated_reps = config.repetitions - self._default_dict['dilation_rate'] // 2
+    for i in range(dilated_reps):
+      self._default_dict['name'] = f'{name}_{i}'
+      x = nn_blocks.DarkResidual(
+          filters=config.filters // scale_filters,
+          filter_scale=residual_filter_scale,
+          **self._default_dict)(
+              x)
+
+    for i in range(dilated_reps, config.repetitions):
+      self._default_dict[
+          'dilation_rate'] = self._default_dict['dilation_rate'] // 2
+      self._default_dict[
+          'name'] = f"{name}_{i}_degrided_{self._default_dict['dilation_rate']}"
```

Comment:
degridded? (probably not worth changing)

Reply:
Fixed